### PR TITLE
fix: loading the twitch token for profile fetches

### LINKED ISSUE #1301: Fetching external data not functioning on v1.13+
the requests out of the machine fail on v1.13+ on production, but not anywhere else

### PR DESCRIPTION
**Pull request by TobiTenno**
### What did you fix? 
closes #1301

- rewrite items, rivens, drops fetches
- create pattern for more caches
- TODOs documented for future caches

to rewrite some of the other fetches, i'll need to rewrite worldstate-emitter w/ fetch & node 18 (probably something ssl-related in json-fetch-cache)

blocks #1305 
blocks #1302 

---

### Reproduction steps

start it up

---

### Considerations
- Does this contain a new dependency? **[No]**
- Does this introduce opinionated data formatting or manual data entry? **[No]**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **[No]**
- Have I run the linter? **[Yes]**
- Is is a bug fix, feature request, or enhancement? **[Bug Fix]**

### CODE REVIEW

**Review by SlayerOrnstein — APPROVED** [review]

**Review by AyAyEm — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "nodemon": "^3.0.3"
   },
   "scripts": {
-    "start": "node main.js",
+    "start": "BUILD=build WFINFO_FILTERED_ITEMS=$WFINFO_FILTERED_ITEMS WFINFO_PRICES=$WFINFO_PRICES node main.js",
```
Comment (AyAyEm) [inline package.json:58]: very hacky me not like

File: src/lib/addons.js [hunk 2]
```diff
@@ -46,10 +48,17 @@ const initSwagger = (app) => {
   app.use(swagger.getMiddleware(swaggConfig));
 };
 
+const initHydration = async () => {
+  await hydrate();
+  const hydration = new CronJob('0 0 * * * *', hydrate, undefined, true);
```
Comment (AyAyEm) [inline src/lib/addons.js:53]: Cron notation is cool but we sometimes forget what it actually means
```suggestion
  // Run every hour
  const hydration = new CronJob('0 0 * * * *', hydrate, undefined, true);
```

Comment (TobiTenno) [reply src/lib/addons.js:54]: i don't entirely agree, but i'm happy to do more docs

File: package.json (hunk 1 above)
Comment (AyAyEm) [reply package.json:58]: well env vars are already in the shell env there shouldn't be a need to do this

Comment (AyAyEm) [reply package.json:58]: or am i missing something?

Comment (TobiTenno) [reply package.json:58]: even if it's duplicative.... why is that hacky? and `BUILD=build` isn't default in the shell (or in context, as i don't generally run these in shell)

Comment (AyAyEm) [reply package.json:58]: well BUILD=build is understandable but ` WFINFO_FILTERED_ITEMS=$WFINFO_FILTERED_ITEMS WFINFO_PRICES=$WFINFO_PRICES` this just seems that you're trying to fix some very specific node issue with the shell environment

Comment (AyAyEm) [reply package.json:58]: as far as my understanding of bash and powershell this really doesnt make sense that's why i called it hacky

Comment (TobiTenno) [reply package.json:58]: it's to parallel the working conditions of the build command.

Comment (AyAyEm) [reply package.json:58]: i see it's hacky then glad you agree with me

**Pull request merged**